pkg/agent/run.go

```diff
@@ -107,7 +107,9 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 		return err
 	}
 
-	util.WaitForAPIServerReady(coreClient, 30*time.Second)
+	if err := util.WaitForAPIServerReady(coreClient, util.DefaultAPIServerReadyTimeout); err != nil {
```

Comment:
Whether 15 minutes or 30 seconds, a wait loop that doesn't honor a `context.Context` in scope at the call site is not acceptable.

Reply:
At the moment, the context we pass around is tied to a signal handler so everything dies/exits when it is cancelled anyway, but I've plumbed it through to both the retry and the request.